tests/e2e/tests/e2e_happy_path/DevWorkspaceHappyPath.spec.ts

```diff
@@ -10,35 +10,185 @@
+            await editor.waitSuggestionWithScrolling(javaFileName, 'run(Class<?> primarySource, String... args) : ConfigurableApplicationContext', 120_000);
+        });
+
+        test('Codenavigation', async () => {
+            await editor.moveCursorToLineAndChar(javaFileName, 32, 17);
+            await editor.performKeyCombination(javaFileName, Key.chord(Key.CONTROL, Key.F12));
+            await editor.waitEditorAvailable(codeNavigationClassName, TimeoutConstants.TS_EDITOR_TAB_INTERACTION_TIMEOUT * 4);
+        });
 });
+    suite('Validation of workspace build and run', async () => {
+        test('Build application', async () => {
+        const taskName: string = 'build';
+        await topMenu.runTask(`${taskName}, ${globalTaskScope}`);
+        await terminal.waitIconSuccess(taskName, 500_000);
+        });
+
+    test('Run application', async () => {
+        const taskName: string = 'run';
+        await topMenu.runTask(`${taskName}, ${globalTaskScope}`);
+        await ide.waitNotification('Process 8080-tcp is now listening on port 8080. Open it ?', 120_000);
+        await ide.clickOnNotificationButton('Process 8080-tcp is now listening on port 8080. Open it ?', 'Open In New Tab');
+       });
+
+    test('Check the running application', async () => {
+        await switchApptWindowAndCheck(SpringAppLocators.springTitleLocator);
+       });
+
+    test('Close the terminal running tasks', async () => {
+        await terminal.rejectTerminalProcess('run');
+        await terminal.closeTerminalTab('run');
+        await warningDialog.waitAndCloseIfAppear();
+        await terminal.closeTerminalTab('build');
+       });
+});
+
+async function checkJavaPathCompletion() {
+    if (await ide.isNotificationPresent('Classpath is incomplete. Only syntax errors will be reported')) {
+        const classpathText: string = fs.readFileSync('./files/happy-path/petclinic-classpath.txt', 'utf8');
+        const workaroundReportText: string = '\n############################## \n\n' +
+            'Known issue: https://github.com/eclipse/che/issues/13427 \n' +
+            '\"Java LS \"Classpath is incomplete\" warning when loading petclinic\" \n' +
+            '\".classpath\" will be configured with next settings: \n\n' +
+            classpathText + '\n' +
+            '############################## \n';
+
+        console.log(workaroundReportText);
+
+        await projectTree.expandPathAndOpenFile(projectName, classPathFilename);
+        await editor.waitEditorAvailable(classPathFilename);
+        await editor.type(classPathFilename, Key.chord(Key.CONTROL, 'a'), 1);
+        await editor.performKeyCombination(classPathFilename, Key.DELETE);
+        await editor.type(classPathFilename, classpathText, 1);
+        await editor.waitTabWithSavedStatus(classPathFilename);
+    }
+}
+
+// when we use devfile v.2 the test app. is opened in the separate window instead of widget
+async function switchApptWindowAndCheck(contentLocator: By) {
```

Comment:
Looks like a typo - redundant "**t**" in method name "switchApp**t**WindowAndCheck()".